### PR TITLE
feat: switch setup.py to console script entry points

### LINKED ISSUE #196: Cannot run `treetime` CLI command on Windows
Hi there

I have installed treetime on windows 11 using:
git clone https://github.com/neherlab/treetime.git
cd treetime
pip install .

I get the message:
Successfully installed phylo-treetime-0.9.1

And I add the folder in my path, but still when I run treetime it cannot find the command. Even when I am in the treetime folder. just wondering if this is a known issue in Window 11?

Thanks
Niloofar

### PR DESCRIPTION
**Pull request by corneliusroemer**
resolves #196 by switching the way `treetime` CLI is defined from `setup` to the recommended `console_entrypoints`. See issue for details.

Notably, this means `treetime` CLI will work properly on windows.

Also resolves some scipy/numpy inconsistency errors by switching the CI to a vanilla `pip` based workflow copied almost 1:1 from Github example workflow for Python. Mixing conda and pip is bound to cause trouble. Oh and it shaves off a minute from each CI run saving resources.

Also gives the more appropriate classifier "stable/maintained" rather than "alpha".

Also deprecates the classifier 3.5 and adds 3.9/10 as these are now tested.

Also moves matplotlib import to after args are passed so that it fails faster

## Testing

Someone should install this on windows and test everything works fine. It works on Linux and macOS. @anna-parker can you try this out on your windows laptop? 

Just need to clone this repo then run `pip install .` then see whether `treetime --help` does what one would expect.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,24 +25,17 @@ jobs:
...
-
-      - run: conda list
-
-      - run: bash test.sh
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest wheel
+      - name: Install treetime
+        run: python -m pip install .
```
Comment (victorlin) [inline .github/workflows/ci.yml:39]: _(non-blocking)_

In Augur, the dev dependencies are [defined in setup.py](https://github.com/nextstrain/augur/blob/67027a6ae6a7a324c8eb646593cc5c0a89650ccf/setup.py#L70-L87) and installed with the package using `pip install ".[dev]"`. I prefer that pattern but this is fine too!

**Review by victorlin — APPROVED** [review]
Tested on Windows 10 + Python 3.7, works great.

```
git clone --single-branch --branch feat--switch-setup.py-to-entry_points -- https://github.com/neherlab/treetime
pip install .\treetime
treetime --help
```

**Comment by corneliusroemer** [comment]
Excellent, thanks for testing! This CI script comes straight from GitHub's
suggestion for vanilla python CI. Don't think there are any dev
dependencies specified so without changing the setup.py further this would
have to stay like this.

**Review by anna-parker — APPROVED** [review]
Hi! Just tested on Windows 10, python 3.9.13 (just went with the default version microsoft suggests currently), it works on the command line, I then manually ran 

`cd test
git clone https://github.com/neherlab/treetime_examples.git 
python run_tests.py `

and all the tests pass, so looks good!

**Pull request merged**